### PR TITLE
Pallet Token Gateway

### PR DESCRIPTION
**Pull request by Wizdave97**
This PR implements a token gateway pallet that allows substrate chains to make asset transfers to and from token gateway EVM deployments.

### CODE REVIEW

File: modules/ismp/pallets/token-gateway-inspector/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn inspect_request(post: &PostRequest) -> Result<(), ismp::Error> {
+			let PostRequest { body, from, source, dest, nonce, .. } = post.clone();
+
+			if source.is_evm() && dest.is_evm() {
+				return Ok(())
+			}
```
Comment (seunlanlege) [inline modules/ismp/pallets/token-gateway-inspector/src/lib.rs:157]: Good idea to add code comments here explaining why this is fine

File: modules/ismp/core/src/error.rs [hunk 2]
```diff
@@ -208,3 +208,20 @@ pub enum Error {
...
+}
+
+impl core::error::Error for Error {}
+
+#[cfg(not(feature = "std"))]
+impl From<Error> for anyhow::Error {
+	fn from(value: Error) -> Self {
+		use alloc::format;
+		use anyhow::anyhow;
+		anyhow!(format!("{value:?}"))
```
Comment (seunlanlege) [inline modules/ismp/core/src/error.rs:225]:
```suggestion
		anyhow::Error::new(value)
```

Comment (seunlanlege) [inline modules/ismp/core/src/error.rs:225]:
```suggestion
		anyhow::Error::new(value)
```

File: modules/ismp/pallets/token-gateway/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+
+use alloc::{string::ToString, vec, vec::Vec};
+use ismp::module::IsmpModule;
+use primitive_types::H256;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+/// Minimum balance for token gateway assets
+const MIN_BALANCE: u128 = 1_000_000_000;
```
Comment (seunlanlege) [inline modules/ismp/pallets/token-gateway/src/lib.rs:52]: Shouldn't this be configurable?

Comment (Wizdave97) [reply modules/ismp/pallets/token-gateway/src/lib.rs:52]: Done

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**